server/src/main/java/org/elasticsearch/action/bulk/TransportAbstractBulkAction.java

```diff
@@ -228,13 +227,10 @@ private boolean applyPipelines(Task task, BulkRequest bulkRequest, Executor exec
             metadata = clusterService.state().getMetadata();
         }
 
-        for (DocWriteRequest<?> actionRequest : bulkRequest.requests) {
```

Comment:
Refactored this, pushing the loop down into IngestService. Did this so that `storedPipelineCache` didn't need to be passed down into `IngestService.resolvePipelinesAndUpdateIndexRequest`. But, this required iterating over the request twice, once withing `IngestService.resolveAndUpdateAllPipelines` and once below.

Reply:
This is an alternative approach without this refactoring: https://github.com/elastic/elasticsearch/pull/116031/files

Reply:
Since this is pretty performance sensitive code, I think it makes sense to avoid the extra looping

Reply:
@dakrone  So you prefer the approach in https://github.com/elastic/elasticsearch/pull/116031/files ?